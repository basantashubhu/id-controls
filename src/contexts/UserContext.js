import {createContext, useContext, useEffect, useState} from "react";
import {auth} from '../firebase'

export const UserContext = createContext(null);

export function useAuth() {
    return useContext(UserContext)
}

export function AuthProvider({children}) {
    const [currentUser, setCurrentUser] = useState()
    const [loading, setLoading] = useState(false)
    const signUp = (email, password) => {
        return auth.createUserWithEmailAndPassword(email, password);
    }
    const logout = () => {
        return auth.signOut()
    }
    const login = (email, password) => {
        return auth.signInWithEmailAndPassword(email, password)
    }
    const resetPassword = (email) => {
        return auth.sendPasswordResetEmail(email)
    }
    useEffect(() => {
        return auth.onAuthStateChanged(user => {
            setCurrentUser(user);
            setLoading(false);
        })
    }, []);
    const value = {
        currentUser,
        signUp,
        logout,
        login,
        resetPassword
    };
    return <UserContext.Provider value={value}>
        {!loading && children}
    </UserContext.Provider>
}