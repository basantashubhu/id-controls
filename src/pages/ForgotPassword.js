import {useState} from "react";
import {useAuth} from "../contexts/UserContext";
import {Link} from "react-router-dom";

const ForgotPassword = () => {
    const {resetPassword} = useAuth()
    const [credential, setCredential] = useState({email: ''})
    const [error, setError] = useState('')
    const [message, setMessage] = useState('')
    const [loading, setLoading] = useState(false)
    const handleChange = (e) => {
        setCredential({...credential, [e.target.name]: e.target.value})
    }
    const handleForgotPassword = async (e) => {
        e.preventDefault();
        try {
            setError('');
            setLoading(true);
            await resetPassword(credential.email, credential.password);
            setMessage('Check your inbox for further information')
            setCredential({email : ''})
        } catch (e) {
            setError('Failed to reset your password');
        } finally {
            setLoading(false)
        }
    }
    return <>
        <div className="container">
            <div className="d-flex justify-content-center">
                <div className="w-100" style={{maxWidth: '400px'}}>
                    <div className="card">
                        <div className="card-body">
                            <h2 className="text-center mb-4">Forgot Password</h2>
                            {error && <div className="alert alert-danger">{error}</div>}
                            {message && <div className="alert alert-success">{message}</div>}
                            <form onSubmit={handleForgotPassword}>
                                <div className="form-group">
                                    <label>Enter your email</label>
                                    <input type="email" name="email" className="form-control" value={credential.email}
                                           onChange={handleChange} required/>
                                </div>
                                <div className="form-group">
                                    <button type="submit" className="btn btn-primary w-100" disabled={loading}>Reset Password</button>
                                </div>
                            </form>
                        </div>
                    </div>
                    <p className="text-center mt-3"><Link to={'/login'}>Login</Link></p>
                </div>
            </div>
        </div>
    </>
}

export default ForgotPassword