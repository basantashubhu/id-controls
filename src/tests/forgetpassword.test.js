import React from 'react'
import {shallow, configure} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import ForgotPassword from '../pages/ForgotPassword'
import * as UserContext from '../contexts/UserContext'
configure({adapter :  new Adapter()})
describe('Test case for forget password', () => {
    let wrapper, useAuth;
    beforeAll(() => {
        useAuth = jest.spyOn(UserContext, 'useAuth')
    });
    beforeEach(() => {
        useAuth.mockClear();
        useAuth.mockReturnValue(() => ({
            resetPassword : () => 1
        }))
    })
    it('email check', () => {
        wrapper = shallow(<ForgotPassword/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('[name="email"]')).toHaveLength(1);
    })
    it('button check', () => {
        wrapper = shallow(<ForgotPassword/>);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('[type="submit"]')).toHaveLength(1);
    })
})
