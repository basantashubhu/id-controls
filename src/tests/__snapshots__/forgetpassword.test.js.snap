// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test case for forget password button check 1`] = `ShallowWrapper {}`;

exports[`Test case for forget password email check 1`] = `ShallowWrapper {}`;
